### PR TITLE
Api get_content - ints zero init bugfix #746

### PR DESCRIPTION
**Pull request by maslenitsa93**

### CODE REVIEW

File: plugins/social_network/social_network.cpp [hunk 1]
```diff
@@ -221,7 +221,9 @@ namespace golos { namespace plugins { namespace social_network {
         if (itr != by_permlink_idx.end()) {
             return get_discussion(*itr, limit);
         }
-        return discussion();
+        auto res = discussion();
```
Comment (afalaleev) [inline plugins/social_network/social_network.cpp]: Move the initialization of an empty discussion to the discussion_helper.
It is more useful, and doesn't require to call the follow::fill_account_reputation() directly from a random place.

Comment (maslenitsa93) [reply plugins/social_network/social_network.cpp]: Couldn't include plugin.hpp (follow) from api (discussion_helper).

File: libraries/api/discussion_helper.cpp [hunk 2]
```diff
@@ -93,6 +96,8 @@ namespace golos { namespace api {
 
 // get_discussion
     discussion discussion_helper::impl::get_discussion(const comment_object& c, uint32_t vote_limit) const {
+        FILE *f1 = fopen("/vite/get_dis.txt", "a");
+        fclose(f1);
```
Comment (afalaleev) [inline libraries/api/discussion_helper.cpp]: Remove these lines.

File: libraries/api/CMakeLists.txt [hunk 3]
```diff
@@ -40,6 +40,7 @@ target_link_libraries(
     golos_protocol
     graphene_utilities
     fc
+    golos::follow
```
Comment (afalaleev) [inline libraries/api/CMakeLists.txt]: Remove this dependency. Library shouldn't depend on plugins.

File: libraries/api/discussion_helper.cpp [hunk 4]
```diff
@@ -204,10 +209,20 @@ namespace golos { namespace api {
     }
 //
 // create_discussion
+    discussion discussion_helper::impl::create_discussion(const std::string& author) const {
+        auto dis = discussion();
+        golos::plugins::follow::fill_account_reputation(database_, author, dis.author_reputation);
```
Comment (afalaleev) [inline libraries/api/discussion_helper.cpp]: The discussion_helper receives pointer to fill_account_account_reputation in constructor.
Replace this with `fill_reputation_(...)`

File: libraries/api/discussion_helper.cpp [hunk 5]
```diff
@@ -1,6 +1,7 @@
 #include <golos/api/discussion_helper.hpp>
 #include <golos/chain/account_object.hpp>
 // #include <golos/plugins/follow/follow_objects.hpp>
+#include <golos/plugins/follow/plugin.hpp>
```
Comment (afalaleev) [inline libraries/api/discussion_helper.cpp]: This `#include` isn't needed.

**Comment by maslenitsa93** [comment]
https://github.com/GolosChain/golos/issues/746

**Pull request merged**